### PR TITLE
Deprecate outdated carousel CSS and views on hourofcode.com

### PR DESCRIPTION
**Pull request by kelbyhawn**
Deprecates outdated CSS styles from various stylesheets around the https://hourofcode.com site and the `carousel_pagination.haml` view since it's no longer used anywhere.

The following PRs updated the places that used to use these styles:
- https://github.com/code-dot-org/code-dot-org/pull/58531
- https://github.com/code-dot-org/code-dot-org/pull/58613
- https://github.com/code-dot-org/code-dot-org/pull/58576

## Links
Jira ticket: [ACQ-1364](https://codedotorg.atlassian.net/browse/ACQ-1364)

## Testing story
Tested locally and searched the codebase for any existing class names that are deleted in this PR (`.carousel-wrapper` is still used with the new Swiper carousels but has been updated [here](https://github.com/code-dot-org/code-dot-org/pull/58531/files#diff-2500d19c34e6f39ac6a60df801c1ee12f5b7d16ed1c47e9ee9ffb4a6b6fb7794)).

Also checked http://localhost.hourofcode.com:3000/us/ai and http://localhost.hourofcode.com:3000/us/beyond to make sure carousels still look as expected.

### CODE REVIEW

**Review by hannahbergam — COMMENTED** [review]
omg I LOVE a deprecation PR. Can you walk me through how you came up with this collection? Are there other PRs that we could link that used to reference these styles?

**Comment by kelbyhawn** [comment]
> omg I LOVE a deprecation PR. Can you walk me through how you came up with this collection? Are there other PRs that we could link that used to reference these styles?

Yes, but I accidentally put this into review too soon 😬 it's waiting on [this PR](https://github.com/code-dot-org/code-dot-org/pull/58613) to get reviewed/merged. I need to check that what's in that PR isn't affected by what's deleted in this PR. What a circus this carousel update has been!

File: pegasus/sites.v3/code.org/styles/040-page.css [hunk 1]
```diff
@@ -493,14 +493,6 @@ th a:active, th a:hover, th a:link, th a:visited {
   color: #c1c1c1 !important
 }
 
-.btn-carousel-slide {
```
Comment (kelbyhawn) [inline pegasus/sites.v3/code.org/styles/040-page.css:496]: These are left over from [the PR](https://github.com/code-dot-org/code-dot-org/pull/58518) that deprecated all of this on the code.org site.

**Comment by kelbyhawn** [comment]
@hannahbergam this is now ready to go! I updated the top description w/ the info you asked for 🙌

**Review by hannahbergam — APPROVED** [review]
LGTM

**Pull request merged**